### PR TITLE
[DeadCode] Skip RemoveUnusedPrivatePropertyRector on property used by Trait in Source

### PR DESCRIPTION
**Pull request by samsonasik**
Currently, the skipping used by trait only works in the same file, if Trait located in different file, eg: in `Source` directory, it not skipped.

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/PhpParser/NodeFinder/PropertyFetchFinder.php [hunk 1]
```diff
@@ -34,34 +50,53 @@ public function findPrivatePropertyFetches(Property | Param $propertyOrPromotedP
...
+
+            if (! $trait instanceof Trait_) {
+                continue;
             }
 
-            $currentClassLike = $node->getAttribute(AttributeKey::CLASS_NODE);
-            return in_array($currentClassLike, $classLikesToSearch, true);
-        });
+            $nodes[] = $trait;
+        }
```
Comment (TomasVotruba) [inline src/PhpParser/NodeFinder/PropertyFetchFinder.php]: Could you extract this logic to AstResolver? Something like `parseClasdReflectionTraits()`?

Comment (samsonasik) [reply src/PhpParser/NodeFinder/PropertyFetchFinder.php]: implemented https://github.com/rectorphp/rector-src/pull/481/commits/cec33c5d2730c63bb738ef8fa8a8e8d302f59196

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**